### PR TITLE
[fix][websocket] Connection header must include not match Upgrade

### PR DESCRIPTION
**Pull request by Rechi**
## Description
RFC 6455 4.1. Client Requirements
The request MUST contain a |Connection| header field whose value MUST include the "Upgrade" token.

## Motivation and Context
newer Firefox versions send `Connection: keep-alive, Upgrade`

## How Has This Been Tested?
Tested with Firefox 56.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/network/websocket/WebSocketV13.cpp [hunk 1]
```diff
@@ -102,7 +104,11 @@ bool CWebSocketV13::Handshake(const char* data, size_t length, std::string &resp
 
   // There must be a "Connection" header with the value "Upgrade"
   value = header.getValue(WS_HEADER_CONNECTION_LC);
-  if (value == NULL || strnicmp(value, WS_HEADER_UPGRADE, strlen(WS_HEADER_UPGRADE)) != 0)
+  if (value == nullptr)
+    elements = std::vector<std::string>();
+  else
+    elements = StringUtils::Split(value, ",");
+  if (elements.empty() || !std::any_of(elements.cbegin(), elements.cend(), [](std::string elem) { return StringUtils::EqualsNoCase(StringUtils::Trim(elem), WS_HEADER_UPGRADE); }))
```
Comment (garbear) [inline xbmc/network/websocket/WebSocketV13.cpp]: can you pass `elem` by reference or use move constructors?

Comment (Rechi) [reply xbmc/network/websocket/WebSocketV13.cpp]: pass by reference doesn't work (StringUtils::Trim doesn't accept const)
`... StringUtils::Trim(std::move(elem)) ...` is the move syntax you mean?

Comment (garbear) [reply xbmc/network/websocket/WebSocketV13.cpp]: yup

**Comment by MartijnKaijser** [comment]
@Montellese if you could have a brief look? If all good we can put it in v17 and do last update.

File: xbmc/network/websocket/WebSocketV13.cpp [hunk 2]
```diff
@@ -102,7 +104,11 @@ bool CWebSocketV13::Handshake(const char* data, size_t length, std::string &resp
 
   // There must be a "Connection" header with the value "Upgrade"
   value = header.getValue(WS_HEADER_CONNECTION_LC);
-  if (value == NULL || strnicmp(value, WS_HEADER_UPGRADE, strlen(WS_HEADER_UPGRADE)) != 0)
+  if (value == nullptr)
+    elements = std::vector<std::string>();
```
Comment (Montellese) [inline xbmc/network/websocket/WebSocketV13.cpp]: Why is this necessary? `elements` has been declared and initialized as `std::vector<std::string>` already.

**Review by Montellese — CHANGES_REQUESTED** [review]
Looks good apart from the unnecessary re-initialization of `elements`.

File: xbmc/network/websocket/WebSocketV13.cpp (hunk 2 above)
Comment (Rechi) [reply xbmc/network/websocket/WebSocketV13.cpp]: just wanted to make it safe for the future, but I'll move elements declaration down here

**Comment by Rechi** [comment]
@Montellese removed the  unnecessary re-initialization of `elements`

**Review by Montellese — APPROVED** [review]
